src/adapter/etl-adapter-parquet/src/Flow/ETL/Adapter/Parquet/ParquetExtractor.php

```diff
@@ -0,0 +1,78 @@
+<?php declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\Parquet;
+
+use function Flow\ETL\DSL\array_to_rows;
+use Flow\ETL\Extractor;
+use Flow\ETL\Filesystem\Path;
+use Flow\ETL\Filesystem\Stream\Mode;
+use Flow\ETL\FlowContext;
+use Flow\ETL\Row\EntryFactory;
+use Flow\ETL\Row\Factory\NativeEntryFactory;
+use Flow\Parquet\ByteOrder;
+use Flow\Parquet\Options;
+use Flow\Parquet\ParquetFile;
+use Flow\Parquet\Reader;
+
+final class ParquetExtractor implements Extractor
+{
+    /**
+     * @param Path $path
+     * @param array<string> $columns
+     */
+    public function __construct(
+        private readonly Path $path,
+        private readonly Options $options,
+        private readonly ByteOrder $byteOrder = ByteOrder::LITTLE_ENDIAN,
+        private readonly array $columns = [],
+        private readonly int $rowsInBatch = 1000,
+        private readonly EntryFactory $entryFactory = new NativeEntryFactory()
+    ) {
+    }
+
+    public function extract(FlowContext $context) : \Generator
+    {
+        foreach ($this->readers($context) as $fileData) {
+            $rows = [];
+
+            foreach ($fileData['file']->values($this->columns) as $row) {
+                if ($context->config->shouldPutInputIntoRows()) {
+                    $rows[] = \array_merge(
+                        $row,
+                        ['_input_file_uri' => $fileData['uri']]
+                    );
+                } else {
+                    $rows[] = $row;
+                }
+
+                if (\count($rows) >= $this->rowsInBatch) {
+                    yield array_to_rows($rows, $this->entryFactory);
+                    $rows = [];
+                }
+            }
+
+            if (\count($rows)) {
```

Comment:
I'm having a little dilemma here. From a code readability perspective, `\count($rows)` sounds to me more natural. It just clearly represents the intention. `[] !== $rows`, on the other hand (at least for me) adds some complexity to the code while reading it. 
If readability would be the only concern, we should use `\count` everywhere but let's look into the performance aspect. 

I prepared a simple benchmark (but maybe I'm missing something).

```php
<?php

// Initialize variables
$iterations = 10000;
$array1 = array_fill(0, 100000, 1);
$array2 = array_fill(0, 100000, 2);

// Start timer for count() method
$start_time1 = microtime(true);

// Run benchmark for count($array) === 0
for ($i = 0; $i < $iterations; $i++) {
    if (\count($array1)) {
        // Do nothing
    }
}

// Stop timer for count() method
$end_time1 = microtime(true);

// Calculate elapsed time for count() method
$elapsed_time1 = $end_time1 - $start_time1;


// Start timer for == [] method
$start_time2 = microtime(true);

// Run benchmark for $array == []
for ($i = 0; $i < $iterations; $i++) {
    if ($array2 !== []) {
        // Do nothing
    }
}

// Stop timer for == [] method
$end_time2 = microtime(true);

// Calculate elapsed time for == [] method
$elapsed_time2 = $end_time2 - $start_time2;

// Output results
echo "Time elapsed for using \\count(\$array): {$elapsed_time1} seconds\n";
echo "Time elapsed for using \$array !== []: {$elapsed_time2} seconds\n";
```

### Results

```console
$ php benchmark_array_count.php
Time elapsed for using \count($array): 0.00012111663818359 seconds
Time elapsed for using $array !== []: 0.00011491775512695 seconds
```

Results do not seem to be impactful enough to sacrifice code readability, based on that my recommendation would be to revert to `\count` as `!== []` is a micro-optimization that does not bring enough value to sacrifice code readability for it.

Reply:
TBH I'm not forcing this approach if we decide it's not worth it, but just using `count()` is less clear than using a clear comparison: `0 === count()` etc

Reply:
> but just using count() is less clear than using a clear comparison: 0 === count() etc

Correct, in this context, `\count()` is used to check if there are remaining rows. Otherwise, it would be `!\count()`. I think it might be possible to use cs-fixer to turn `!\count()` into `\count() === 0` or the other way around. 

For now, I would revert from comparing arrays with `[]` to `count`.